crates/key-value-redis/src/lib.rs

```diff
@@ -53,14 +53,14 @@ struct RedisStore {
 
 #[async_trait]
 impl Store for RedisStore {
-    async fn get(&self, key: &str) -> Result<Vec<u8>, Error> {
+    async fn get(&self, key: &str) -> Result<Option<Vec<u8>>, Error> {
         let mut conn = self.connection.lock().await;
         let result: Vec<u8> = conn.get(key).await.map_err(log_error)?;
 
         if result.is_empty() {
-            Err(Error::NoSuchKey)
+            Ok(None)
         } else {
-            Ok(result)
+            Ok(Some(result))
```

Comment:
It looks like this will return `None` for empty values. Is that what we want here or should this instead be just:

```rust
   conn.get(key).await.map_err(log_error)
}
```

Reply:
Great catch! The previous behavior was wrong it seems.